### ISSUE TITLE
Fails on noisy input

### ISSUE THREAD

**Issue by troxel** [Knio/pynmea2#55]
I get a failure trying to access a property on a bad input... Looks like checksum is checked only if it exists in string. If no checksum is in the string the package will still try to parse but fail when trying to access properties. For example attached file will fail when trying to access the properties. 

[test_bad_rmc.py.txt](https://github.com/Knio/pynmea2/files/565613/test_bad_rmc.py.txt)

Version 1.5.3

**Comment by Knio** [comment]
It's unclear what the expected behavior is here when you're passing it corrupt input.

I added a strict checking argument to `parse()` that will enforce that the data has a valid checksum, if you want to fast-fail, and I also suppressed the exception coming from the automatic type conversion so that it is consistent and will just return the corrupt data field, similar to how string (or missing) fields behave already.

Let me know if this resolves your issue!
